addons/mail/models/mail_channel.py

```diff
@@ -381,15 +381,26 @@ def add_members(self, partner_ids=None, guest_ids=None, invite_to_rtc_call=False
                 if post_joined_message:
                     # notify existing members with a new message in the channel
                     if member.partner_id == self.env.user.partner_id:
-                        notification = _('<div class="o_mail_notification">joined the channel</div>')
+                        notification = _(
+                            '<div class="o_mail_notification">%(member)s joined the channel</div>',
+                            member=current_partner.name or current_guest.name,
+                        )
```

Comment:
Maybe you shouldn't insert the author's name in the notification, because it prevents the name to be updated when the member is renamed.

I think you should rather send the notification with its placeholders, then fill them on the client side, so that it avoids the name being frozen at the value it had at creation in database. Why fill them on the client side specifically? My hope is that having notifications in a consistent form (i.e. without the placeholders filled with variable stuff) would allow us to use them as keys for translation, and thus have a translation that depends on the user's locale rather than the company's locale.

Reply:
We cannot do this correctly and properly right now, because notification messages are `mail.message`, and their content is essentially the same for everyone.

It's quite hard to deal with message content with these notification messages, so I wouldn't advice storing a template and do the processing in JS on them. Instead, we should use simpler data structure for these notification messages, and not just reuse `mail.message`.

Not that easy to do for now, hence keeping `mail.message` and be aware of current limitation.